torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -481,16 +481,18 @@ Tensor solve_backward_A(const Tensor & grad, const Tensor & self, const Tensor &
   return -at::matmul(grad_self, solution.conj().transpose(-2, -1));
 }
 
-Tensor cumsum_backward(const Tensor & x, int64_t dim) {
-  // Need to check numel to see if there are no values (such as shape [0,2], and dim to see if x is a scalar.
-  if (x.dim() == 0 || x.numel() == 0) {
-    return x;
-  }
-  auto ret = at::cumsum(-x, dim);
-  auto ret_sum = ret.narrow(dim, ret.size(dim) - 1, 1).clone(at::MemoryFormat::Preserve);
-  ret -= ret_sum.expand(ret.sizes());
-  ret += x;
-  return ret;
+Tensor cumsum_backward(const Tensor & grad, int64_t dim) {
+  /* Implements w.flip(dim).cumsum(dim).flip(dim) without copying.
+     This implementation gives a similar performance on GPU but a
```

Comment:
Once we merge this PR, this comment won't make sense because the impl we are referencing here wouldn't exist anymore so I think we should either remove this comment or add a reference `x% speedup on GPU w.r.t., approach abc"

Reply:
I'll just delete this.